python/mxnet/test_utils.py

```diff
@@ -1538,3 +1546,53 @@ def discard_stderr():
     finally:
         os.dup2(old_stderr, stderr_fileno)
         bit_bucket.close()
+
+@contextmanager
+def np_random_seed(seed=None):
+    """
+    Runs a code block with a new state of np.random.
+    To impose rng determinism, invoke e.g. as in:
+
+    with np_random_seed(1234):
+        ...
+
+    To impose rng non-determinism, invoke as in:
+
+    with np_random_seed():
+        ...
+
+    """
+
+    try:
+        saved_rng_state = np.random.get_state()
+        np.random.seed(seed)
+        yield
+    finally:
+        # Reinstate prior state of np.random
+        np.random.set_state(saved_rng_state)
+
+# Set seed and output to stderr (to avoid default nosetests filtering of stdout)
+def set_np_random_seed(seed=None, ostream=sys.stderr):
+    """Set the np.random seed and announce the value to an output stream
+
+    Parameters
+    ----------
+
+    seed: int
+        Seed to pass to np.random.seed().  Should be None to set and output
+        a randomly chosen value.
+    ostream :
+        Stream to announce the new seed value to.
+
+    The expected use of this function is to set the seed globally before
+    a suite of tests and output the set value to help reproduce any failures
+    that are dependent on the random data.  To fix the seed for a single test
+    without modifying the randomness of subsequent tests in the same file,
+    use 'np_random_seed'.
+    """
+    if seed is None:
+        seed = np.random.randint(0, np.iinfo(np.uint32).max)
+    if ostream is not None:
+        ostream.write('Setting np.random seed to %s.\n' % seed)
```

Comment:
Why not use logging? I think is better to writing to stdout or stderr in general.